refresh/index.js

```diff
@@ -673,74 +671,59 @@ module.exports = generators.Base.extend({
+    if (!shouldGenerateClient && !shouldGenerateServer) return;
 
-    function geniOS(callback) {
-      swaggerize.parse.call(self, self.fromSwagger)
-      .then((response) => {
-        performSDKGeneration.call(self, self.appname + '_iOS_SDK', 'ios_swift', JSON.stringify(response.loaded), function (generatedID) {
-          getiOSSDK.call(self, self.appname + '_iOS_SDK', generatedID, function() {
-              self.itemsToIgnore.push('/' + self.appname + '_iOS_SDK*');
-              callback();
-          });
+    this.log(chalk.green('Generating SDK(s) from swagger file(s)...'));
+    var generationTasks = [];
+    if (shouldGenerateClient) generationTasks.push(generateClientAsync.call(this));
+    if (shouldGenerateServer) generationTasks.push(generateServerAsync.call(this));
+    return Promise.all(generationTasks);
+
+    function generateClientAsync() {
+      var clientSDKName = `${this.projectName}_iOS_SDK`;
+      return performSDKGenerationAsync(clientSDKName, 'ios_swift', JSON.stringify(this.loadedApi))
+        .then(generatedID => getClientSDKAsync(clientSDKName, generatedID))
+        .then(() => {
+          this.itemsToIgnore.push(`/${clientSDKName}*`);
         });
-      })
-      .catch(function(e) {
-        done(e);
-      });
     }
 
-    function genServer(callback) {
-      var numFinished = 0;
-      for (var index = 0; index < self.serverSwaggerFiles.length; index++) {
-        
-        swaggerize.parse.call(self, self.serverSwaggerFiles[index])
-        .then((response) => {
-          if (response.loaded['info']['title'] == undefined) {
-            self.env.error(chalk.red('Could not extract title from Swagger API.'));
-          } else {
-            var sdkName = response.loaded['info']['title'].replace(/ /g, '_') + '_ServerSDK';
-            performSDKGeneration.call(self, sdkName, 'server_swift', JSON.stringify(response.loaded), function(generatedID) {
-              getServerSDK.call(self, sdkName, generatedID, function() {
-                numFinished += 1;
-                if(numFinished === self.serverSwaggerFiles.length) {
-                  callback();
+    function generateServerAsync() {
+      return Promise.map(this.serverSwaggerFiles, file => {
+        return swaggerize.parse(this.fs, file)
+          .then(response => {
+            if (response.loaded.info.title === undefined) {
+              this.env.error(chalk.red('Could not extract title from Swagger API.'));
+            }
+
+            var serverSDKName = response.loaded.info.title.replace(/ /g, '_') + '_ServerSDK';
+            return performSDKGenerationAsync(serverSDKName, 'server_swift', JSON.stringify(response.loaded))
+              .then(generatedID => getServerSDKAsync(serverSDKName, generatedID))
+              .then(sdk => {
+                this.sdkTargets.push(serverSDKName);
+                if (sdk.packages.length > 0) {
+                  this.sdkPackages = sdk.packages; // FIXME! Seems bad
```

Comment:
Seems like this only takes the packages from the latest SDK that was generated (provided it had any dependencies at all). Do all server SDKs always have the same dependencies? Or should we really be merging and de-duplicating them?